src/utils/shallowEqual.js

```diff
@@ -3,7 +3,9 @@ export default function shallowEqual(objA, objB) {
     return true
   }
 
-  if (objA === undefined || objB === undefined) {
+  if (objA === undefined || objB === undefined ||
```

Comment:
if `objA` is `undefined` then `typeof objA` won't be `'object'` so there's no need to check for both, just remove the `=== undefined` check